tests/validation/tests/v3_api/test_encryption_secrets.py

```diff
@@ -0,0 +1,41 @@
+from .common import *  # NOQA
+import pytest
+from .test_rke_cluster_provisioning import rke_config
+
+
+def test_create_custom_cluster_with_encryption():
+    # Adds custom encryption to the RKE config
+    secret_encryption_config = {"secrets_encryption_config": {"enabled":
+                                True, "custom_config": {}}}
+    custom_resources = {"resources": "secrets", "providers": {
+                        "aescbc": {"keys":
+                                   {"name": "k-fw5hn", "secret":
+                                    "RTczRjFDODMwQzAyMDVBREU4NDJBMUZFNDhCNzM5N0I="}},
+                        "identity": {}}}
+    custom_config = {"apiVersion": "apiserver.config.k8s.io/v1",
+                     "kind": "EncryptionConfiguration", "resources": custom_resources}
+    secret_encryption_config["secrets_encryption_config"]["custom_config"].update(custom_config)
+
+    custom_encryption_rke_config = rke_config
+    custom_encryption_rke_config["services"]["kube-api"] = secret_encryption_config
+
+    aws_nodes = \
+        AmazonWebServices().create_multiple_nodes(
+            3, random_test_name("jenkins-custom-secret-encryption"))
+    node_roles = [["controlplane"], ["etcd"],
+                  ["worker"], ["worker"], ["worker"]]
+    client = rancher.Client(url=CATTLE_TEST_URL + "/v3",
+                            token=USER_TOKEN, verify=False)
+    cluster = client.create_cluster(
+        name=random_name(),
+        driver="rancherKubernetesEngine",
+        rancherKubernetesEngineConfig=custom_encryption_rke_config)
+    assert cluster.state == "provisioning"
+    i = 0
+    for aws_node in aws_nodes:
+        docker_run_cmd = \
+            get_custom_host_registration_cmd(
+                client, cluster, node_roles[i], aws_node)
+        aws_node.execute_command(docker_run_cmd)
+        i += 1
+    validate_cluster(client, cluster)
```

Comment:
We still need to validate the custom encryption has been set ?

Reply:
Addressed in changes.
Validation is done by checking `/etc/kubernetes/ssl/encryption.yaml`